src/ScheduledTask/FetchPaymentMethodLogosHandler.php

```diff
@@ -0,0 +1,180 @@
+use Adyen\Shopware\PaymentMethods\PaymentMethods;
+use Adyen\Shopware\Service\ConfigurationService;
+use Exception;
+use Psr\Log\LoggerAwareTrait;
+use Shopware\Core\Checkout\Payment\PaymentMethodEntity;
+use Shopware\Core\Content\Media\File\MediaFile;
+use Shopware\Core\Content\Media\MediaService;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\MessageQueue\ScheduledTask\ScheduledTaskHandler;
+use Symfony\Component\HttpFoundation\Request;
+
+class FetchPaymentMethodLogosHandler extends ScheduledTaskHandler
+{
+    use LoggerAwareTrait;
+
+    /**
+     * @var ConfigurationService
+     */
+    private $configurationService;
+    /**
+     * @var MediaService
+     */
+    private $mediaService;
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $paymentMethodRepository;
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $mediaRepository;
+    /**
+     * @var bool
+     */
+    private $enableUrlUploadFeature;
+
+    public function __construct(
+        EntityRepositoryInterface $scheduledTaskRepository,
+        ConfigurationService $configurationService,
+        MediaService $mediaService,
+        EntityRepositoryInterface $paymentMethodRepository,
+        EntityRepositoryInterface $mediaRepository,
+        bool $enableUrlUploadFeature = true
+    ) {
+        parent::__construct($scheduledTaskRepository);
+        $this->configurationService = $configurationService;
+        $this->mediaService = $mediaService;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+        $this->mediaRepository = $mediaRepository;
+        $this->enableUrlUploadFeature = $enableUrlUploadFeature;
+    }
+
+    public static function getHandledMessages(): iterable
+    {
+        return [ FetchPaymentMethodLogos::class ];
+    }
+
+    public function run(): void
+    {
+        if (!$this->enableUrlUploadFeature) {
+            $this->logger->debug('Configuration `shopware.media.enable_url_upload_feature` is disabled.');
+            return;
+        }
+
+        $context = Context::createDefaultContext();
+        $environment = $this->configurationService->getEnvironment();
+        foreach (PaymentMethods::PAYMENT_METHODS as $identifier) {
+            /** @var PaymentMethodInterface $paymentMethod */
+            $paymentMethod = new $identifier();
+
+            // Look up corresponding payment_method entity.
+            $result = $this->paymentMethodRepository->search(
+                (new Criteria())->addFilter(new EqualsFilter(
+                    'handlerIdentifier',
+                    $paymentMethod->getPaymentHandler()
+                )),
+                $context
+            );
+            if ($result->getTotal() === 0) {
```

Comment:
Is it correct like this?
```suggestion

            // Skip if the payment method is not registered in ShopWare
            if ($result->getTotal() === 0) {
```

Reply:
Yes